wakepy/methods/wsl.py

```diff
@@ -0,0 +1,100 @@
+import enum
+from abc import ABC, abstractmethod
+from subprocess import Popen, PIPE
+from typing import List
+
+from wakepy.core import DbusAdapter, Method, ModeName, PlatformName
+
+class Flags(enum.StrEnum):
+    KEEP_RUNNING = "$ES_CONTINUOUS -bor $ES_SYSTEM_REQUIRED"
+    KEEP_PRESENTING = "$ES_CONTINUOUS -bor $ES_SYSTEM_REQUIRED -bor $ES_DISPLAY_REQUIRED"
+    RELEASE = "$ES_CONTINUOUS"
+
+
+class WslSetThreadExecutionState(Method, ABC):
+    """This is a method which calls the SetThreadExecutionState function from
+    the kernel32.dll using PowerShell.exe. The SetThreadExecutionState informs the system that it
+    is in use preventing the system from entering sleep or turning off
+    the display while the application is running" (depending on the used
+    flags)."""
+
+    # This should support Windows 7 and above out of the box.  Windows XP and above if PowerShell 2 is installed.
+    supported_platforms = (PlatformName.WSL,)
+
+    def __init__(self, dbus_adapter: DbusAdapter | None = None):
+        self._powershell: Popen = None
+
+        super().__init__(dbus_adapter)
+
+    def enter_mode(self):
+        self._powershell = Popen(["powershell.exe"], stdout=PIPE, stderr=None, stdin=PIPE, universal_newlines=True)
+
+        # C# Code to allow PowerShell to call SetThreadExecutionState
+        code = """$code = @'
+[DllImport("kernel32.dll", CharSet = CharSet.Auto, SetLastError = true)]
+public static extern void SetThreadExecutionState(uint esFlags);
+'@""".strip().split("\n")
+        
+        # Write the code line, by line
+        self._write(*code)
+        # Write a blank line to complete the code block.
+        self._write("")
+
+        # Add the code above to the PowerShell Context.
```

Comment:
This was the line I had question in the initial version, so I suggest we add your answer as comment :D  something like:

```
Some C# code to add the `code` above to the PowerShell Context. Add-Type call sets it up as something we can call within PowerShell.
```

?